pyneuroml/swc/LoaderSWC.py

```diff
@@ -0,0 +1,110 @@
+class Point:
+    def __init__(self, x, y, z, diameter):
+        self.x = float(x)
+        self.y = float(y)
+        self.z = float(z)
+        self.diameter = float(diameter)
+
+
+class Segment:
+    def __init__(self, segment_id, name=None):
+        self.id = int(segment_id)
+        self.name = name or f"Segment_{segment_id}"
+        self.proximal = None
+        self.distal = None
+        self.parent_id = None
+        self.parent = None
+        self.children = []
+
+    def set_proximal(self, x, y, z, diameter):
+        self.proximal = Point(x, y, z, diameter)
+
+    def set_distal(self, x, y, z, diameter):
+        self.distal = Point(x, y, z, diameter)
+
+
+class SegmentGroup:
+    def __init__(self, group_id, name, neurolex_id=None):
+        self.id = group_id
+        self.name = name
+        self.neurolex_id = neurolex_id
+        self.members = set()
+
+    def add_member(self, segment_id):
+        self.members.add(segment_id)
+
+
+class Morphology:
+    NEUROLEX_IDS = {
+        "soma": "sao1044911821",
+        "axon": "sao1369643253",
+        "dendrite": "sao1211023249",
+        "apical_dendrite": "sao285106870",
+        "basal_dendrite": "sao735526996",
+    }
+
+    SWC_TO_GROUP = {
+        1: ("soma", "Soma"),
+        2: ("axon", "Axon"),
+        3: ("dendrite", "BasalDendrite"),
+        4: ("dendrite", "ApicalDendrite"),
+    }
+
+    def __init__(self):
+        self.segments = {}
+        self.segment_groups = {}
+        self.root = None
+
+    def add_segment(self, segment):
+        self.segments[segment.id] = segment
+        if segment.parent_id == -1:
+            self.root = segment
+        elif segment.parent_id in self.segments:
+            segment.parent = self.segments[segment.parent_id]
+            segment.parent.children.append(segment)
+
+    def create_segment_groups(self):
```

Comment:
this is good for a start, but there are going to be a lot of more segment groups in the cell. Each unbranched section of the morphology is a segment group. All of these will be of one of the standard types: soma or dendrite or axon. So, we also need to detect all the SWC sections that form an unbranched chain and put those in a segment group. See the `add_unbranched_segment_group` helper function in the neuroml.Cell class:

https://libneuroml.readthedocs.io/en/latest/userdocs/coreclasses.html#neuroml.nml.nml.Cell.add_unbranched_segment_group

also:

https://libneuroml.readthedocs.io/en/latest/userdocs/coreclasses.html#neuroml.nml.nml.Cell.add_unbranched_segments